node/src/neighborhood/gossip_acceptor.rs

```diff
@@ -1658,7 +1642,7 @@ mod tests {
         System::current().stop();
         system.run();
         assert_eq!(Qualification::Matched, qualifies_result);
-        assert_eq!(handle_result, GossipAcceptanceResult::Ignored,);
+        assert_eq!(handle_result, GossipAcceptanceResult::Absorbed);
```

Comment:
How is that possible that you say Absorbed, which I consider different enough from Ignored, and despite that the database data did not change? Are the assertions below right and complete? Why do we get None for the Node which sent the Gossip and wanted us to know that his IP is different. Strangely enough, we won't find it at all in our database when we run our assertions.

The same with the stream handler pool, I'd expect some activity. 

If the origin only guy just contacted us with this special delivery we've got them waiting at the doorsteps. They've already initiated a connection, different from the previous one, and so we can accept and hold it.

Reply:
Actually, the database _did_ change (we changed the NodeAddr of the transmitting Node), but it's not a change that should trigger a Gossip broadcast.

Remember, this change is because of a comment you made in the last pass.

Reply:
I'm still not so calm about the assertions, and don't know if you thought about them. 

Why does the record contain no NodeAddr? I'm coming to an explanation that according to the real world it should not contain it also before the test fires off. If yes, then I don't understand your point saying that we changed the NodeAddr of the Node which sent this to us. If it should have been None before (because it also was running in this same mode) and is None now still, then we did not change any NodeAddr.

Also, what do you take from my question whether we actually consider this guy an ongoing full neighbor of ours because we just assured ourselves about having a TCP connection with the guy (if not dropped from them - anyway, we can design it so that they won't do it) and so we know all unknowns, eh, which is just the address, they also initiated this connection, so we don't break rules that they want to be OriginateOnly and so we can start using a route through this updated link. 

That's why I was thinking why is that we don't make the stream handler pool involved. Because, not knowing all details, I suppose it participate in some kind of reestablishment of the existing stream keys previously created for the contacting Node.